src/jit/importer.cpp

```diff
@@ -11226,6 +11226,20 @@ void Compiler::impImportBlockCode(BasicBlock* block)
 #endif // _TARGET_64BIT_
                 }
 
+#ifdef FEATURE_CORECLR
+                if (IsOperArithmetic(oper))
+                {
+                    const typeInfo& tiOp1 = impStackTop(1).seTypeInfo;
+                    const typeInfo& tiOp2 = impStackTop().seTypeInfo;
+
+                    // check for structure too because typeof(ldloca struct) is struct
```

Comment:
I don't see how the type of ldloca(struct) can be a struct, it would have to be a byref,

And in any case I don't think we want to fold the special case for struct (if is is needed) into the TYP_LONG case.
Make it a separate BADCODE check.